lib/puppet/resource_api/simple_provider.rb

```diff
@@ -16,6 +16,8 @@ def set(context, changes)
 
         should = change[:should]
 
+        raise 'SimpleProvider cannot be used with a Type that is not ensurable' unless context.type.ensurable?
+
```

Comment:
Please pull the actual PDK-924 change out into its own commit. And then PR the `context.type` change as a separate PR, so it gets its own CHANGELOG entry. This is important enough to warrant that.